lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -49,16 +49,19 @@ def compute_queue_time(env)
         end
 
         def call(env)
+          # Find out if this is rack within rack
+          previous_request_span = env[Ext::RACK_ENV_REQUEST_SPAN]
```

Comment:
Does Rack actually get called recursively? It's very 👻 if so, because the Rack instrumentation does stuff like this:
https://github.com/DataDog/dd-trace-rb/blob/93cd757770df59455fc94d04bb2db62f3dd22ada/lib/ddtrace/contrib/rack/middlewares.rb#L123

Reply:
Nesting whole rack (or sinatra) apps within rack (or sinatra, or rails) apps is definitely possible, and actually tested by the specs. Without this kind of fetch/check/restore (at `call`'s `ensure`), consequences appeared in the spec, e.g distributed tracing headers being mistakenly picked up by the deeper rack integration middleware, severing them from their actual parent.

The mentioned reset line I catered for when based on 0.x, but it appears it does not exist anymore on 1.0, the corresponding setter at `call` beginning became `Datadog::Tracing.continue_trace!(trace_digest)`.

I surmised the following comment above the line you quoted was no longer relevant as this may now better handled by the new trace concepts:

```
          # TODO: Remove this once we change how context propagation works. This
          # ensures we clean thread-local variables on each HTTP request avoiding
          # memory leaks.
```

But do tell me if I am mistaken as these concepts are very new to me.